### PR TITLE
moved compatibility matrix to upgrade guide

### PR DESCRIPTION
**Pull request by SwastikTripathi**
As mentioned in #6044 the Compatibility Matrix should also be present in Upgrade Guide.
This will aid in easier understanding of changes introduced.

### CODE REVIEW

File: docs/source/upgrade.rst [hunk 1]
```diff
@@ -360,3 +360,26 @@ Changed Behavior of count_nonzero Function
...
+| v6   | 3.0~7.x | 8.0~10.1       | n/a      | n/a      | 1.3~2.4        | 5.0~7.5 | 2.7 & 3.4~3.8 | 1.9~1.17  | (not specified) | (not specified)        | v6.7.0 |
++------+---------+----------------+----------+----------+----------------+---------+---------------+-----------+-----------------+------------------------+--------+
+| v5   | 3.0~7.x | 8.0~10.1       | n/a      | n/a      | 1.3~2.4        | 5.0~7.5 | 2.7 & 3.4~3.7 | 1.9~1.16  | (not specified) | (not specified)        | v5.4.0 |
++------+---------+----------------+----------+----------+----------------+---------+---------------+-----------+-----------------+------------------------+--------+
+| v4   | 3.0~7.x | 7.0~9.2        | n/a      | n/a      | 1.3~2.2        | 4.0~7.1 | 2.7 & 3.4~3.6 | 1.9~1.14  | (not specified) | (not specified)        | v4.5.0 |
++------+---------+----------------+----------+----------+----------------+---------+---------------+-----------+-----------------+------------------------+--------+
+(*1) CUDA Compute Capability (*2) Hightly experimental support with limited features
```
Comment (toslunar) [inline docs/source/upgrade.rst:385]: Could you use the footnote markup?
https://docutils.sourceforge.io/docs/user/rst/quickref.html#footnotes

Comment (SwastikTripathi) [reply docs/source/upgrade.rst:385]: Sure.

**Comment by kmaehashi** [comment]
Thanks! Could you rewrite the markdown table into reST format? I'd suggest using `list-table` format for maintainability:
https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table

**Comment by SwastikTripathi** [comment]
Working on it :)

**Pull request closed without merging**